curriculum/challenges/english/08-coding-interview-prep/project-euler/problem-35-circular-primes.english.md

```diff
@@ -25,12 +25,12 @@ tests:
     testString: assert(circularPrimes(100) == 13, '<code>circularPrimes(100)</code> should return 13.');
   - text: <code>circularPrimes(100000)</code> should return 43.
     testString: assert(circularPrimes(100000) == 43, '<code>circularPrimes(100000)</code> should return 43.');
-  - text: <code>circularPrimes(250000)</code> should return 45.
-    testString: assert(circularPrimes(250000) == 45, '<code>circularPrimes(250000)</code> should return 45.');
-  - text: <code>circularPrimes(500000)</code> should return 49.
-    testString: assert(circularPrimes(500000) == 49, '<code>circularPrimes(500000)</code> should return 49.');
-  - text: <code>circularPrimes(750000)</code> should return 49.
-    testString: assert(circularPrimes(750000) == 49, '<code>circularPrimes(750000)</code> should return 49.');
+  - text: <code>circularPrimes(250000)</code> should return 43.
```

Comment:
It's not correct. It should be 45. Your solution does not count 193939 and 199933.

Reply:
@ValeraS I understand that but other rotations of `193939` i.e. `939391` are not in range of `0-250000` therefore it shouldn't be counted.

I checked other solutions online such as this one https://blog.dreamshire.com/project-euler-35-solution/
For `circularPrimes(250000)` they also return 43 and so on.

Reply:
and `939391` is not counted, but IMO `193939` should. The solution from the blog solves particular task for `n = 1000000`. And this solution is correct for `n = 10^k`

Reply:
If you change `n = 250000`, you'll see it outputs `43`, likewise for `n = 500000 // 43` and `n = 750000 // 43`.
Another [solution](https://gist.github.com/Anivarth/40d2d708418f7abdc06c9938c178aac7) which returns `43` for `n = 250000`. Here's the [working](https://www.onlinegdb.com/BkxAPr_2m) solution

I really think we should only be checking for primes in the range of the given input or the problem changes drastically from users expectation.

Reply:
is `193939` circular prime and less than `250 000`? In the task, all circular primes are counted independently. For example, `17` is one circular prime and `71` is another and we count both for `n > 100`.

Reply:
Yes, but at the same time its other rotations arent in the range ex: 939391. 

This is the ambiguity in the problem itself so I can see where you're coming from. 

We should come to a consensus about if we want to consider  `193939` as a circular prime if `n = 250000`. If we do consider it, then I think we should add a note saying the circular prime's rotation can exceed the input provided.

That being said - I think we shouldn't consider `193939` as a circular prime speaking from the sake of the tests because then the algorithm would also check if `939391` is also a prime. 
Assuming the `isPrime()` is of complexity **O(n)** (could be refactored to be **O(sqrt(n))**) and then we have inputs in the range of `100,000 - 1,000,000`. So any solution will certainly take a bit more time.

You can decide if we should go with that or not. I'll make the changes accordingly.

Reply:
Reverted the tests, fixed solution and added a note in the description.
Let me know if it requires any further changes.